spec4d-alloy/README.md

```diff
@@ -0,0 +1,89 @@
+satisfiability solvers.
+
+## Experimental Setup
+
+Since the SAT solver SAT4j shipped with Alloy is not very performant, we run the
+experiments with an award-winning SAT solver called [Kissat][] by Armin Biere et
+al. To reproduce the experiments, you have to build Kissat from sources by
+following the installation instructions.
+
+To generate a CNF file -- which a SAT solver is able to consume:
+
+ 1. Open an experiment file such as [ffg-exp1][] in the Alloy IDE.
+ 
+ 1. Check `Options`/`Solver: Output CNF to file`/`Output CNF to file`.
+ 
+ 1. Generate the file via `Execute`/`Run noAccountableSafety...`.
+ 
+ 1. Copy the generated file in a convenient location. We accompany every .als
+    file with the generated .cnf file.
+
+Finally, run the SAT solver against the file:
+
+```sh
+$ kissat --unsat ffg-exp1.cnf
+```
+
+The expected output is:
+
+```
+s UNSATISFIABLE
+```
+
+This means that the property `noAccountableSafety` does not have a model.
+Hence, we were not able to find a counterexample to `AccountableSafety`.
+
+## Experimental Results
+
+Similar to the experiments with a [direct SMT encoding][smt-enc] for CVC5, we
+conduct experiments for the same kinds of inputs:
+
+| Input      | #blocks | #checkpoints | #signatures | #ffg_votes | #votes | runtime  | memory  |
+|------------|--------:|-------------:|------------:|-----------:|-------:|---------:|--------:|
+| [ffg-exp1] |    3    |      5       |      4      |      5     |   12   |  4 sec   |  35 MB  |
+| [ffg-exp2] |    4    |      5       |      4      |      5     |   12   | 10 sec   |  40 MB  |
+| [ffg-exp3] |    5    |      5       |      4      |      5     |   12   | 15 sec   |  45 MB  |
+| [ffg-exp4] |    3    |      6       |      4      |      6     |   15   | 57 sec   |  52 MB  |
+| [ffg-exp5] |    4    |      6       |      4      |      6     |   15   | 167 sec  |  55 MB  |
+| [ffg-exp6] |    5    |      6       |      4      |      6     |   15   | 245 sec  |  57 MB  |
+| [ffg-exp7] |    6    |      6       |      4      |      6     |   15   | 360 sec  |  82 MB  |
+| [ffg-exp8] |    5    |      7       |      4      |      6     |   24   | 1h 27m   |  156 MB |
+
+In addition to the above experiments, we ran a few experiments that would have
+the inputs of the size comparable to those produced by our TLA+ specification:
+
+| Input       | #blocks | #checkpoints | #signatures | #ffg_votes | #votes | runtime | memory  |
+|-------------|--------:|-------------:|------------:|-----------:|-------:|--------:|--------:|
+| [ffg-exp10] |    3    |      15      |      4      |      5     |   12   | 31 sec  | 56 MB   |
+| [ffg-exp11] |    4    |      20      |      4      |      5     |   12   | 152 sec | 94 MB   |
+| [ffg-exp12] |    5    |      25      |      4      |      5     |   12   | 234 sec | 117 MB  |
+| [ffg-exp13] |    7    |      15      |      4      |      10    |   40   | >10 days | 300 MB  |
```

Comment:
Can you add a short sentence or two after this table explaining why we think the jump from "a few seconds" to "multiple days" happens when the votes increase as such. I would mention the math we did for the smallest # of votes you need to observe finalized conflicting blocks.